src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -235,11 +235,18 @@ public static JsonNode convertToJson(Schema schema, Object logicalValue) {
     try
     {
       final Schema.Type schemaType;
-      if (schema == null)
-      {
-        schemaType = ConnectSchema.schemaType(value.getClass());
-        if (schemaType == null)
-          throw SnowflakeErrors.ERROR_5015.getException("Java class " + value.getClass() + " does not have corresponding schema type.");
+      if (schema == null) {
+        Schema.Type primitiveType = ConnectSchema.schemaType(value.getClass());
+        if (primitiveType != null) {
+          schemaType = primitiveType;
+        } else {
+          if (value instanceof java.util.Date) {
+            schema = Timestamp.SCHEMA;
```

Comment:
We also use `java.util.Date` to represent `Date.SCHEMA` and `Time.SCHEMA` values as well.
A mix of schema'd `Time.SCHEMA/Date.SCHEMA` objects with schemaless `java.util.Date` objects would cause this to switch between strings formatted with `TIME_FORMAT/ISO_DATE_FORMAT` and timestamps written as numbers. But without knowing the schema assigned to a different record, the connector will just need to go along with it.

I think this is probably the best option given the circumstances, and is just a caveat for anyone running in the strange mixed schema/schemaless environment.

Reply:
Yeah--since this is only triggered at all when there isn't any schema associated with the given value, there's no way to distinguish between date, time, and timestamp; I opted to use timestamp in order to avoid losing information.